vllm/model_executor/models/fuyu.py

```diff
@@ -237,6 +237,7 @@ def __init__(self,
             self.image_feature_size,
             config.hidden_size,
             quant_config=quant_config,
+            gather_output=True,
         )
         self.language_model = PersimmonForCausalLM(config,
```

Comment:
```suggestion
        self.language_model = PersimmonForCausalLM(config.text_config,
```

While you're here, can you fix this as well? I'm working on #7168 and failed to load the model because of this.

Reply:
The new HF config will automatically populate `text_config` from the `config.json` on HF repo.

Reply:
```py,diff
--- a/vllm/model_executor/models/fuyu.py
+++ b/vllm/model_executor/models/fuyu.py
@@ -23,7 +23,7 @@ import torch
 import torch.nn as nn
 import torch.utils.checkpoint
 from PIL import Image
-from transformers import FuyuConfig, FuyuImageProcessor
+from transformers import FuyuConfig, FuyuImageProcessor, PersimmonConfig
 
 from vllm.attention import AttentionMetadata
 from vllm.config import CacheConfig, MultiModalConfig
@@ -227,10 +227,12 @@ class FuyuForCausalLM(nn.Module, SupportsMultiModal):
                  quant_config: Optional[QuantizationConfig] = None) -> None:
         super().__init__()
         self.config = config
+
         self.multimodal_config = multimodal_config
+        text_config: PersimmonConfig = config.text_config
 
         self.padding_idx = config.pad_token_id
-        self.vocab_size = config.text_config.vocab_size
+        self.vocab_size = text_config.vocab_size
         self.image_token_id = _IMAGE_TOKEN_ID
         self.image_feature_size = config.patch_size**2 * config.num_channels
 
@@ -239,7 +241,7 @@ class FuyuForCausalLM(nn.Module, SupportsMultiModal):
             config.hidden_size,
             quant_config=quant_config,
         )
-        self.language_model = PersimmonForCausalLM(config,
+        self.language_model = PersimmonForCausalLM(text_config,
                                                    cache_config=cache_config,
                                                    quant_config=quant_config)
         self.make_empty_intermediate_tensors = (
diff --git a/vllm/model_executor/models/persimmon.py b/vllm/model_executor/models/persimmon.py
index f05a22a0..c4fef759 100644
--- a/vllm/model_executor/models/persimmon.py
+++ b/vllm/model_executor/models/persimmon.py
@@ -217,7 +217,8 @@ class PersimmonModel(nn.Module):
                  quant_config: Optional[QuantizationConfig] = None,
                  prefix: str = ""):
         super().__init__()
-        self.vocab_size = config.text_config.vocab_size
+
+        self.vocab_size = config.vocab_size
 
         self.embed_tokens = VocabParallelEmbedding(config.vocab_size,
                                                    config.hidden_size)
@@ -265,19 +266,20 @@ class PersimmonModel(nn.Module):
 class PersimmonForCausalLM(nn.Module):
 
     def __init__(self,
-                 config,
+                 config: PersimmonConfig,
                  cache_config: Optional[CacheConfig] = None,
                  quant_config: Optional[QuantizationConfig] = None):
         super().__init__()
+
         self.config = config
-        self.vocab_size = config.text_config.vocab_size
+        self.vocab_size = config.vocab_size
         self.model = PersimmonModel(config,
                                     cache_config=cache_config,
                                     quant_config=quant_config)
-        self.lm_head = ParallelLMHead(config.text_config.vocab_size,
+        self.lm_head = ParallelLMHead(config.vocab_size,
                                       config.hidden_size,
                                       bias=False)
-        self.logits_processor = LogitsProcessor(config.text_config.vocab_size)
+        self.logits_processor = LogitsProcessor(config.vocab_size)
         self.sampler = Sampler()
         self.make_empty_intermediate_tensors = (
             self.model.make_empty_intermediate_tensors)
```

I have this patch but it fails the TP test (which is to be fixed by your PR)

Reply:
Done.